versioned_docs/version-stable/nym-simple-websocket-tutorial.md

```diff
@@ -0,0 +1,658 @@
+npm --version
+```
+
+
+### Installing Typescript and Bundling our Application
+
+Inside your terminal, type and enter:
+```
+npm install -g typescript
+```
+This will install typescript for us globally. The -g directive tells npm to install the package in the global shared node_modules folder usually at the root folder in your computer
+
+After this , go ahead and create a new folder to start your project in and give it an appropriate name eg; ‘Mixnet Websocket Project’. Continue to then do the following:
+
+1. Open your terminal in the folder you created (or by `cd`'ing to the directory), type and enter:
+
+    ```
+    npm init
+    ```
+    The following chunk of output will then be presented to you. The terminal will prompt your to provide some input for the the sections name to  license. You can simply just press enter after each prompt (like the example below) and it will just set default values. Alternatively, you can go ahead and fill out each value as you wish. 
+
+    <img src="/img/tutorials/simple-websocket/image1.png"/>
+
+    You will then notice a `package.json` file has been created in the directory:
+
+    <img src="/img/tutorials/simple-websocket/image2.png"/>
+
+ 2. Continuing with our terminal, type and enter:
+    
+    ```
+    npm install typescript
+    ```
+    After this point , we should open up our chosen IDE (VSCode, Sublime Text, etc) and open up the folder we are working in (the one with the `package.json`).
+
+    <img src="/img/tutorials/simple-websocket/image3.png"/>
+
+    We can see that typescript has been added to our `dependencies`. Typescript is now in our project.
+
+
+ 3. Back in our terminal, type and enter:
+    
+    ```
+    npm install ts-node --save-dev
+    ```
+    This package (`ts-node`) allows us to build a typescript application in a node environment.
+
+ 4. Create a new file in the same level as `package.json` called `tsconfig.json` Inside that file , copy and paste the code below into it.
+
+    ```
+    {
+        "compilerOptions": {
+            "module": "commonjs",
+            "esModuleInterop": true,
+            "target": "es6",
+            "moduleResolution": "node",
+            "sourceMap": true,
+            "outDir": "dist"
+        },
+        "lib": ["es2015"]
+    }
+    ```
+ 5. Now that we have got to this point, we want to be able to run our application on localhost to make sure everything is working okay. 
+    We also want to be able to work on our application while its running and make sure it automatically picks up any saved changes when we view it on the browser. To do this , we are going to use the Parcel bundler.
+
+    Back in your terminal, type and enter:
+
+    ```
+    npm install --global parcel-bundler
+    ```
+
+    After the npm install has finished, create a new folder called src in the same level as our tsconfig.json and package.json (our project root). Inside there , create 2 new files. One called `index.html` and one called `index.ts`.
+
```

Comment:
Add the output of a command like `tree` or something here, showing what your directory structure should look like printed in the console. Just so people know they have the correct code project structure before diving into coding